common/serde-codegen/src/main.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) The Libra Core Contributors
```

Comment:
for purely detecting format changes, this seems unnecessary? Perhaps put it into a second PR? It is also worth exploring if we want this in the Libra code base.

Reply:
Let me close this PR and create a new one without the codegen.